### ISSUE TITLE
ARIA grid, roving tabindex

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar#6527]
A ["roving tabindex"](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_roving_tabindex) is when a component accepts focus on only the subset of its UI that is "selected". Once the component has been focused, the user can move the selection cell(s)/row(s) to other parts of the component's UI using the arrow keys typically.

Examples (you must tab into the component with your keyboard first):
- [mini spreadsheet](https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/dataGrids.html)
- [mini calendars](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Grid_Role#examples)

## Notes for each view

- daygrid
    - `role="grid"`
        - multi-day events are focusable for each day
        - after the last event of the day, or after the more-events link, have a dummy focasable line that instructions how to move to other days/weeks. this is also useful for preventing the user from accidentally moving past the last event to the next tabbable element on the page
        - multi-tier headers for vertical resource view will follow [w3c recommendation]()
        - must press enter/space on more-link. hiding the more-link popover will function similarly to [w3c disclosure example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/disclosure/disclosure-navigation.html)
    - roving keyboard shortcuts
        - left/right - forward/back a day. depends on dir
        - up/down - if multi-week view, forward/back a week. otherwise, does nothing
        - home/end - move to first/last cell in row
        - page-up/page-down move to first/last cell in whole view
- timegrid
    - `role="grid"`
        - only a single row of columns. the all-day section and the time-axis are considered the same "cell" for roving
        - similar to daygrid, after the last event of the day, have a dummy focusable line that instructions how to move to other days
        - all-day section will follow description of daygrid
        - multi-tier headers will follow description of daygrid
        - focusing "more-link" behaves same as daygrid
    - roving keyboard shortcuts
        - left/right - forward/back a day. depends on dir
        - up/down - move forward/backward in time-of-day. default selected "time" is the all-day section
        - home/end - move to first/last day
        - page-up - move to the all-day section of the current day. if allDaySlot set to false, go to first time slot
        - page-down - move to last time slot of the current day.
- timeline
    - `role="grid"`
        - roving will only happen along the left-to-right timeslots, as a vertical line (aka "cursor")
    - roving keyboard shortcuts
        - left/right - move cursor left/right along time axis
- resource-timeline
    - `role="treegrid"`
        - reminder: resource-timeline can have nested resources and resource groups. they expand
        - reminder: resource-timeline can have data cells via resourceAreaColumns
        - reminder: resourceAreaColumns can have grouping, resulting in cells that have multi-row spans
            - they are not considered part of any row
            - they are considered row "headers"
        - as stated in the [table-based views ticket](https://github.com/fullcalendar/fullcalendar/issues/6526), the event cell for each row will not share a common DOM ancestor, thus tabbing order will be messed up. We might need to intercept and redirect the tab focus, similar to what the [w3c modal dialog example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html) does.
        - the user cannot select the event "cell" as a whole. rather, as the user moves laterally left/right through the time axis
    - keyboard shortcuts: similar to what the [treegrid w3c example](https://www.w3.org/TR/wai-aria-practices-1.1/examples/treegrid/treegrid-1.html) dictates, with each slot on the time axis treated as its own cell.
    - will we need [aria-readonly](https://www.w3.org/TR/wai-aria-1.1/#aria-readonly) on nearly every resourceAreaColumns cell?
- list view
    - will not do roving

### Open Questions

- after this is implemented, set the entire fullcalendar's [role to "application"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Application_Role) ?
- should we make each day's events part of an [aria-compatible list](https://www.scottohara.me/blog/2018/05/26/aria-lists.html)? will setting role="application" on the root element affect how screen readers interpret this?
